discord/role.py

```diff
@@ -79,70 +135,128 @@ class RoleTags:
         "_premium_subscriber",
         "_available_for_purchase",
         "_guild_connections",
+        "bot_id",
+        "_data",
     )
 
     def __init__(self, data: RoleTagPayload):
-        self.bot_id: int | None = _get_as_snowflake(data, "bot_id")
-        self.integration_id: int | None = _get_as_snowflake(data, "integration_id")
-        self.subscription_listing_id: int | None = _get_as_snowflake(
+        self._data: RoleTagPayload = data
+        self.integration_id: int | None = _parse_tag_int(data, "integration_id")
+        self.subscription_listing_id: int | None = _parse_tag_int(
             data, "subscription_listing_id"
         )
-        # NOTE: The API returns "null" for each of the following tags if they are True, and omits them if False.
-        # However, "null" corresponds to None.
-        # This is different from other fields where "null" means "not there".
-        # So in this case, a value of None is the same as True.
-        # Which means we would need a different sentinel.
-        self._premium_subscriber: Any | None = data.get("premium_subscriber", MISSING)
-        self._available_for_purchase: Any | None = data.get(
-            "available_for_purchase", MISSING
+        self.bot_id: int | None = _parse_tag_int(data, "bot_id")
+        self._guild_connections: bool | None = _parse_tag_bool(
+            data, "guild_connections"
+        )
+        self._premium_subscriber: bool | None = _parse_tag_bool(
+            data, "premium_subscriber"
+        )
+        self._available_for_purchase: bool | None = _parse_tag_bool(
+            data, "available_for_purchase"
         )
-        self._guild_connections: Any | None = data.get("guild_connections", MISSING)
 
-    def is_bot_managed(self) -> bool:
-        """Whether the role is associated with a bot."""
+    @property
+    def is_bot_role(self) -> bool:
+        """Whether the role is associated with a bot.
+        .. versionadded:: 2.7
+        """
         return self.bot_id is not None
 
-    def is_premium_subscriber(self) -> bool:
-        """Whether the role is the premium subscriber, AKA "boost", role for the guild."""
-        return self._premium_subscriber is None
+    @property
+    def is_booster_role(self) -> bool:
+        """Whether the role is the "boost", role for the guild.
+        .. versionadded:: 2.7
+        """
+        return self._guild_connections is False and self._premium_subscriber is True
+
+    @property
+    def is_guild_product_role(self) -> bool:
+        """Whether the role is a guild product role.
+
+        .. versionadded:: 2.7
+        """
+        return self._guild_connections is False and self._premium_subscriber is False
 
+    @property
     def is_integration(self) -> bool:
         """Whether the guild manages the role through some form of
         integrations such as Twitch or through guild subscriptions.
         """
         return self.integration_id is not None
```

Comment:
This should be deprecated as it does not narrow down the role to a specific "type"

Reply:
Something else to be added for YouTube twitch etc...